api/proto/teleport/legacy/types/types.proto

```diff
@@ -6735,6 +6735,8 @@ message AWSMatcher {
   // KubeAppDiscovery controls whether Kubernetes App Discovery will be enabled for agents running on
   // discovered clusters, currently only affects AWS EKS discovery in integration mode.
   bool KubeAppDiscovery = 8 [(gogoproto.jsontag) = "kube_app_discovery,omitempty"];
+  // SetupAccessForARN is the role that the discovery service should create EKS Access Entries for.
+  string SetupAccessForARN = 9 [(gogoproto.jsontag) = "setup_access_for_arn,omitempty"];
```

Comment:
I think we can give more detailed comment, as it's a tricky thing. As I understand we do this to be able to split Kube Service and Discovery service, which can run under different ARNs, we probably should mention it.
Also, maybe we can add some special value (`self` or `callerIdentity` etc) which would just set it up for the caller identity? Since Kube and Discovery often run from one process/one ARN, so it's easier for user to configure this.

Reply:
please re-check

Reply:
I think we can still make comment a bit more detailed mentioning that it's Discovery service setting up for Kube service.